coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelToSqlNodeConverter.java

```diff
@@ -92,8 +92,36 @@ public boolean requireCastOnString() {
       }
 
       /**
-       * Override this method so that there will be explicit and correct table alias for selected fields, which is necessary for
-       * data type derivation on SqlNodes
+       * Override this method so that there will be explicit table alias for all the fields, which is necessary for
+       * data type derivation on struct fields.
+       *
+       * For the following input SQL:
+       * CREATE TABLE default.complex(s struct(name:string, age:int));
+       * SELECT split(complex.s.name, ' ') name_array
+       * FROM default.complex
+       * WHERE complex.s.age = 25;
+       *
+       * The input RelNode is:
+       * LogicalProject(name_array=[split($0.name, ' ')])
+       *   LogicalFilter(condition=[=($0.age, 25)])
+       *     LogicalTableScan(table=[[hive, default, complex]])
+       *
+       * With this override, the converted SqlNode is:
+       * SELECT `split`(`complex`.`s`.`name`, ' ') AS `name_array`
+       * FROM `default`.`complex` AS `complex`
```

Comment:
I'm also just realizing that the javadoc doesn't justify the change `FROM `default`.`complex`` -> `FROM `default`.`complex` AS `complex`

Could you please add a 1 liner to elaborate why & how the `AS` clause was added here?

Reply:
Thanks for suggestion, added in the end, it's just a side effect of this override.